jdaviz/components/tray_plugin.vue

```diff
@@ -13,22 +13,44 @@
       <span> {{ getDisabledMsg() }}</span>
     </v-row>
     <div v-else>
+      <v-row v-if="uses_active_status">
+        <v-switch
+          v-model="keep_active"
+          @change="$emit('update:keep_active', $event)"
+          label="keep active"
+          hint="consider plugin active (showing any previews and enabling all keypress events) even when not opened"
+          persistent-hint>
+        </v-switch>
+      </v-row>
       <slot></slot>
     </div>
   </v-container>
 </template>
 
 <script>
 module.exports = {
-  props: ['disabled_msg', 'description', 'link', 'popout_button'],
+  props: ['disabled_msg', 'description', 'link', 'popout_button',
+          'uses_active_status', 'plugin_ping', 'keep_active'],
   methods: {
     isDisabled() {
       return this.getDisabledMsg().length > 0
     },
     getDisabledMsg() {
       return this.disabled_msg || ''
     },
-  }
+    sendPing() {
+      if (!this.$el.isConnected) {
+        return
+      }
+      this.$emit('update:plugin_ping', Date.now())
+      setTimeout(() => {
+        this.sendPing()
+      }, 200)  // ms
+    }
+  },
+  mounted() {
+    this.sendPing();
+  },
```

Comment:
this is where any instance of the UI sends pings back to python (by updating the `plugin_ping` traitlet with the current timestamp) at regular intervals until it is no longer connected.